pulser/simulation/simconfig.py

```diff
@@ -0,0 +1,136 @@
+            solver_options (qutip.Options): options for the qutip solver.
+        """
+    def __init__(self, noise: list[NOISE_TYPES] = [],
+                 runs: int = 15, samples_per_run: int = 5,
+                 temperature: float = 50., laser_waist: float = 175.,
+                 solver_options: qutip.Options = qutip.Options(max_step=5)):
+        self.noise = noise
+        self.temperature = temperature
+        self.runs = runs
+        self.samples_per_run = samples_per_run
+        self.laser_waist = laser_waist
+        self.solver_options = solver_options
+        self.init_spam()
+
+    def __str__(self) -> str:
+        s = ""
+        s += "Options:\n"
+        s += "----------       \n"
+        s += "Noise types:       " + str(self.noise) + "\n"
+        s += "Spam dictionary:   " + str(self.spam_dict) + "\n"
+        s += "Temperature:       " + str(self.temperature) + "K" + "\n"
+        s += "Number of runs:    " + str(self.runs) + "\n"
+        s += "Samples per runs:  " + str(self.samples_per_run) + "\n"
+        s += "Laser waist:       " + str(self.laser_waist) + "μm \n"
+        s += "\n" + "Solver Options: \n"+str(self.solver_options)[10:-1]+"\n"
+        return s
+
+    @property
+    def noise(self) -> list[NOISE_TYPES]:
+        return self._noise
+
+    @noise.setter
+    def noise(self, noise_types: abc.Sequence[NOISE_TYPES]) -> None:
+        self._noise: list[NOISE_TYPES] = []
+        for noise_type in noise_types:
+            self._add_noise(noise_type)
+
+    def _add_noise(self, noise_type: NOISE_TYPES) -> None:
+        """Adds a noise model to the SimConfig instance.
+
+            Args:
+                noise_type (NOISE_TYPES): Choose among:
+                -   'dephasing': Random phase (Z) flip
+                -   'doppler': Noisy doppler runs
+                -   'amplitude': Noisy gaussian beam
+                -   'SPAM': SPAM errors. Adds:
+                    --  eta: Probability of each atom to be badly prepared
+                    --  epsilon: Probability of false positives
+                    --  epsilon_prime: Probability of false negatives.
+        """
+        # Check proper input:
+        if noise_type not in get_args(NOISE_TYPES):
+            raise ValueError('Not a valid noise type')
+        self._noise.append(noise_type)
+
+    @property
+    def spam_dict(self) -> dict[str, float]:
+        return self._spam_dict
+
+    def set_spam(self, eta: Optional[float] = None,
+                 epsilon: Optional[float] = None,
+                 epsilon_prime: Optional[float] = None) -> None:
+        """Allows the user to change SPAM parameters in dictionary"""
+        values = {'eta': eta, 'epsilon': epsilon,
+                  'epsilon_prime': epsilon_prime}
+        for param in values:
+            val = values[param]
+            if val is not None:
+                if val > 1. or val < 0.:
+                    raise ValueError('Invalid value : must be between 0 and 1')
```

Comment:
In trying to keep the error messages more uniform and informative, try to follow the guidelines laid out in #185.